### PR TITLE
Add get_indexed_range to LiteMap

### PR DESCRIPTION
**Pull request by sffc**

### CODE REVIEW

File: experimental/asciitrie/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# asciitrie [![crates.io](https://img.shields.io/crates/v/asciitrie)](https://crates.io/crates/asciitrie)
```
Comment (sffc) [inline experimental/asciitrie/README.md:1]: whoops, need to remove this file from this PR

**Review by Manishearth — DISMISSED** [review]

**Review by Manishearth — DISMISSED** [review]

File: utils/litemap/src/map.rs [hunk 2]
```diff
@@ -199,6 +199,66 @@ where
...
+    ///
+    /// let mut map = LiteMap::new_vec();
+    /// map.insert(1, "one");
+    /// map.insert(2, "two");
+    ///
+    /// let borrowed_map = map.as_sliced();
+    /// assert_eq!(sub_map.get(&1), Some(&"one"));
+    /// assert_eq!(sub_map.get(&2), Some(&"two"));
+    /// ```
+    pub fn as_sliced(&self) -> LiteMap<K, V, &S::Slice> {
```
Comment (sffc) [inline utils/litemap/src/map.rs:251]: @Manishearth What should this function be called?

Comment (Manishearth) [reply utils/litemap/src/map.rs:251]: i think it's fine

**Comment by sffc** [comment]
I had pulled this out of #2722, but I'm going to close this because I may not need it any more.

**Pull request closed without merging**